source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -205,7 +213,13 @@ class ProcessorState : public Logger::Loggable<Logger::Id::ext_proc> {
   // Flag to track whether Envoy already received the new timeout message.
   // Envoy should receive at most one such message in one particular state.
   bool new_timeout_received_{false};
+
+  // The buffer which stores all the received data that had not been shipped.
+  Buffer::OwnedImpl received_data_;
```

Comment:
I looks to me like it would be better to encapsulate `received_data_` in the `ChunkQueue` class.

Reply:
good idea